File: DataAlgo_forPy/Search_Sort/bubble_sort.js
// Implement bubble sort

const buble_sort = (array) => {
  let notSorted = true

  while (notSorted) {
    let swapped = false
    for (var i = 0; i < array.length - 1; i++) {
      if (array[i] > array[i + 1]) {
        swapped = true

        let temp = array[i + 1]
        array[i + 1] = array[i]
        array[i] = temp

      }
    }
    if (!swapped) { notSorted = false}
  }

  return array
}

console.log(buble_sort([5,4,3,2,1])) // [1,2,3,4,5]

File: DataAlgo_forPy/Search_Sort/bubble_sort.rb
# Implement bubble sort

def buble_sort(array)
  not_sorted = true

  while not_sorted
    swapped = false

    (0...array.length - 1).each do |i|
      if array[i] > array[i + 1]
        swapped = true
        array[i], array[i + 1]  = array[i + 1], array[i]
      end
    end

    not_sorted = false unless swapped
  end

  array
end

p buble_sort([5,4,3,2,1]) # [1,2,3,4,5]
